pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -571,6 +577,40 @@ private List<PinotTaskConfig> createPinotTaskConfigs(List<SegmentZKMetadata> sel
     return pinotTaskConfigs;
   }
 
+  private Map<String, String> getPushTaskConfig(Map<String, String> batchConfigMap, String downloadUrls) {
+    try {
+      String[] downloadURLList = downloadUrls.split(MinionConstants.SEGMENT_NAME_SEPARATOR);
+      if (downloadURLList.length > 0) {
+        String downloadUrl = downloadURLList[0];
+        URI downloadURI = URI.create(downloadUrl);
+        URI outputDirURI = null;
+        if (!downloadURI.getScheme().contentEquals("http")) {
```

Comment:
Maybe I misunderstand something, but I don't fully follow the logic here. Why is the outputDir decided by the downloadUrl?

Reply:
Because the segment needs to be copied to some place that is accessible by the controller in case of metadata push.
If user doesn't set the outputDirUri then the best place would be to copy it in deep store. The download url refers to deep store path and hence using it to derive the metadata.

Reply:
If that's the goal, then we just need to use `controllerConfig.getDataDir()/tableName`. What do you think?

Reply:
That makes more sense.

Reply:
Added the commit to use dataDir.